Flax implementation of DPT

younesbelkada opened the pull request:
# What does this PR do?

I tried to implement DPT (Dense Prediction with Transformers) in Flax during my free time! 🚀 
By the way it is the first Segmentation and Depth Estimation model implemented in Flax!

Nits/TODOs:
- [x] Figure out how to properly call `BatchNorm` and `Dropout` inside a `Sequential` 
- [ ] Test equivalency tests 
- [ ] Write documentation - For now they're just copy/pasted
Quetions:
- Why the loss is not implemented in `modeling_dpt.py` ? I can probably help on that since I have already implemented the loss for a university project: https://github.com/antocad/FocusOnDepth/blob/master/FOD/Loss.py 
 
cc @NielsRogge @sanchit-gandhi @patil-suraj

younesbelkada committed:
first commit:

- added DPT in Flax
- all non slow tests passes in local
- still some nits have to be investigated

younesbelkada committed:
make quality

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_17770). All of your documentation changes will be reflected on that endpoint.

ArthurZucker commented on src/transformers/models/dpt/modeling_flax_dpt.py:
```diff
@@ -0,0 +1,1121 @@
+"""
+
+
+class FlaxPatchEmbeddings(nn.Module):
```
Missing `#copied from FlaxViT` as I think it is copy pasted

ArthurZucker reviewed (commented):
Gonna have a deeper look later on

ArthurZucker committed:
implement pt_flax equivlence test to bypass  `AttributeError: 'NoneType' object has no attribute 'tolist'`

younesbelkada committed:
add few changes:

- BN seems to work now
- Equivalency test pass with tol=1e-4 but only with a hack
